python/mage/node_classification/utils/extract_from_database.py

```diff
@@ -0,0 +1,262 @@
+import torch
+from torch_geometric.data import Data, HeteroData
+import mgp
+import numpy as np
+from tqdm import tqdm
+import random
+from collections import Counter
+import torch_geometric.transforms as T
+import typing
+
+def extract_from_database(
+    ctx: mgp.ProcCtx,
+    train_ratio: float,
+    features_name: str,
+    class_name: str,
+) -> typing.Tuple[HeteroData, str, typing.Dict, typing.Dict]:
```

Comment:
This is such a long function. Split it in logical parts. For example edge fetching, nodes fetching, graph creation

Reply:
Thank you, great advice!